tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -9748,6 +9752,22 @@ clang::QualType HLSLExternalSource::CheckVectorConditional(
     return QualType();
   }
 
+  // TBD: Handle LValues instead of always doing RValue cast.
+  VK = VK_RValue;
+  // TBD: Use OK_BitField and OK_VectorComponent when appropriate?
+  OK = OK_Ordinary;
+
+  // Cast condition to RValue
+  if (Cond.get()->isLValue())
+    Cond.set(CreateLValueToRValueCast(Cond.get()));
```

Comment:
One thing to note about the existing code is that if Cond is dependent (and LHS and RHS are not), it will return Cond type as the expression type.  I don't think this is correct, but that's not the bug being fixed in this PR.